### ISSUE TITLE
Consistent interface: ActiveIndex

### ISSUE THREAD

**Issue by nikolasrieble** [recharts/recharts#3962]
Currently a few components offer to programatically set the activeIndex, which enables our users to customise Tooltip behaviour and force showing the Tooltip without a mouse over. 

We should extend this to all components and unify the API. 

Goal: 
- [ ] Identify all components that should have activeIndex 
- [ ] Implement activeIndex for Line 
- [ ] ...

<img width="321" alt="Screenshot 2023-11-14 at 21 25 13" src="https://github.com/recharts/recharts/assets/27995606/e9926293-d427-495d-a88f-fa3c880bbadd">

**Comment by nikolasrieble** [comment]
@andrewangelle You recently worked on this, would you consider picking this up as well?

**Comment by andrewangelle** [comment]
@nikolasrieble Yea, sure. I'll comment back here with a proposed list of components before I start implementing.

**Comment by andrewangelle** [comment]
@nikolasrieble I'm seeing Line, Area, Bar, Radar as needing activeIndex added to their api's. Agree?

Also, in regards to:
> which enables our users to customise Tooltip behaviour and force showing the Tooltip without a mouse over.

I wouldn't say it forces showing the tooltip without a mouse over. It forces showing the active shape/dot/bar without the Tooltip. Right? Or am I mis-understanding the ask?

**Comment by nikolasrieble** [comment]
Indeed, it does not force showing the Tooltip. But if you force showing the Tooltip (on the Tooltip itself https://github.com/recharts/recharts/pull/3935) you need a way to also set the activeIndex - currently partially lacking. 
So, yes, it uses the activeShape, and passes the activeIndex on to the parent state. That would be the ask.

**Comment by wmy94827** [comment]
> 我看到 Line、Area、Bar、Radar 需要将 activeIndex 添加到他们的 api 中。同意？
> 
> 此外，关于：
> 
> > 这使我们的用户能够自定义工具提示行为，并在不使用鼠标的情况下强制显示工具提示。
> 
> 我不会说它会在没有鼠标的情况下强制显示工具提示。它强制在没有工具提示的情况下显示活动形状/点/条。右？还是我误解了这个问题？

+1

**Comment by MadsBuchmann** [comment]
Just to chime in with a use-case for this where it would be very useful. 

I'm currently working on a dashboard for an application, with multiple charts showing different data for the same months in three different charts. Whenever a user hovers one of the months, i would like the tooltip to appear for the month on all charts. Then the user has all of the information about the month they need, without having to hover each bar individually. 

If there's anyway i can help out with implementing this let me know :)!

**Comment by ckifer** [comment]
@MadsBuchmann I think what you want can be accomplished by adding a syncId to each of your charts and using the tooltip synchronization feature
